src/middlewares/requestingInterceptorMiddleware.ts

```diff
@@ -0,0 +1,39 @@
+import { Dispatch, Middleware, MiddlewareAPI } from 'redux';
+import IStore from '../models/IStore';
+import IAction from '../models/IAction';
+import { delay } from '../utilities/delay';
+
+const actionTypeTimeStampMap: Map<string, number> = new Map();
+
+const requestingInterceptorMiddleware: Middleware = (store: MiddlewareAPI<Dispatch, IStore>) => (next: Dispatch) => async (
+  action: IAction<any>
+): Promise<void> => {
+  const isRequestType: boolean = action.type.includes('REQUEST_');
+  const isFinishedRequestType: boolean = action.type.includes('_FINISHED');
+  const isStartRequestType: boolean = action.type.includes('REQUEST_') && !isFinishedRequestType;
+
+  if (isStartRequestType) {
+    actionTypeTimeStampMap.set(action.type, new Date().getTime());
```

Comment:
use `performance.now()` instead because you're dealing with _timing_ not _time_.

Reply:
Not sure `performance.now()` would work for this. If two actions started before the first actions finishes how would you know the correct timing?

Reply:
Maybe I don't know how `performance.now()` works.